metrics/registry.go

```diff
@@ -45,21 +53,19 @@ type Registry interface {
 
 	// Unregister the metric with the given name.
 	Unregister(string)
-
-	// Unregister all metrics.  (Mostly for testing.)
-	UnregisterAll()
 }
 
-// The standard implementation of a Registry is a mutex-protected map
+// The standard implementation of a Registry uses sync.map
 // of names to metrics.
 type StandardRegistry struct {
-	metrics map[string]interface{}
-	mutex   sync.Mutex
+	metrics sync.Map
+	// not representing 1:1 map size.
+	size atomic.Int32
```

Comment:
Is there a need for this? The only use I could see (quick glance) was to set the initial size of a map, which is just ann optimization that can be dropped IMO

Reply:
Please remove the size